### ISSUE TITLE
Parameter select: queries do not distinguish between synonyms

### ISSUE THREAD

**Issue by gcthys** [neo4j-labs/neodash#241]
I have a database which contains a large number of synonyms: people with the same name, e.g 3x John Doe.
Parameter Select automatically creates a parameter by concatenating "neodash", the Node Label and the Property Name, e.g. "$neodash_name_text".
This works fine when the properties are unique. But when they are not, queries produce information of all synonyms (in my case: of all 3 John Does). 
Please offer an option to concatenate "neodash", the Node Label and the Node Identity instead. In my example: "$neodash_name_identity".
Thank you.

**Comment by BennuFire** [comment]
Hi @gcthys !

I don't get what's the problem here. All the property names are unique per label, so this variable name identifies unequivocally each Label-Property combination. Close words become a problem due to the fact that are shown in the list. If they are the same, you always have the **Advanced Setting** Deduplicate Suggestion Values.

**Comment by gcthys** [comment]
Hello @BennuFire ,
Thanks for taking the time to look at my challenge.
I will try and explain myself better, in plain English.
Suppose I have the following persons in the database:
p1: Person1 with the "preferred" n1.name "Hieronymus III Verdussen" who has a name variant n3.name "Hieronymus Verdussen" and is born on 01-01-1650.
p2: Person2 with the "preferred" n2.name "Hieronymus IV Verdussen" who also has a name variant n4.name "Hieronymus Verdussen" and is born on 01-01-1675.
(this is sort of a real example: there is a lineage of 7 persons in the database who bear this same name; the Roman numerals are only added by scholars to keep them apart but they never appear in original documents)
When I search for "Hieronymus Verdussen" I get to choose between the two alternatives: the synonyms n3 and n4 which are linked to p1 and p2 resp.
Suppose I choose Person1, then the parameter $neodash_name_name becomes "Hieronymus Verdussen"
Any subsequent query will use $neodash_name_name and find n1 and n2 and link them to all the data related to p1 and p2 resp. As a consequence the report will list 2 birth dates, whereas it should only show 01-01-1650.
This would be solved if the parameter would not be based upon the searched field (name) but on the ID of the selected field.
Please let me know if this explains. If not, I will record a screen video. Thanks.

**Comment by BennuFire** [comment]
Ah ok!

This is completely different. Long story short, you would like to set a $neodash variable to a value different than the one displayed. With the Parameter Select chart, that's something that's on our backlog. Lemme consider it for 2.2.1 or 2.2.2

**Comment by gcthys** [comment]
Indeed, that would solve the problem.
Looking forward to the new releases!

**Comment by gcthys** [comment]
Hi @BennuFire ,
I have checked the new Parameter Selector, but I can't figure out what to do in order to solve my problem.

Let me reiterate.
I have a database of persons, among whom several dozens with identical names. I cannot disambiguate them because they are real persons. So there would be e.g. two different nodes n:Name:
n.NamedID: N1234   n.Name: John Doe
n.NamedID: N9999   n.Name: John Doe

In the Parameter Selector I set the node to Name and the property to n.Name and it displays the two persons in the pull-down. Fine, this is exactly what needs to happen.

But when I select one name (e.g. the one with ID 1234), $neodash_name_name is set to "John Doe" and in all the following reports the data on both N1234 and N9999 are being returned because both persons respond to the same parameter $neodash_name_name.

So, after having selected an n.Name, I need to set the parameter to $neodash_name_nameid in order to return only the data of N1234.

The new Property Display Name Override does not do this: it lets me change $neodash_name_name into $neodash_name_nameid, but then the selector looks for NamedID instead of on Name and no John Does are found.

Sorry for being a slow learner, but can you please explain how to do this (or complement the User Guide)?
Thank you.

**Comment by BennuFire** [comment]
Hi @gcthys !

No worries at all. With the new parameter selector you should be able to enable _Property Display Name Override_ right on the advanced settings. 

![image](https://user-images.githubusercontent.com/7397411/217265675-f55139fb-d1d2-4cca-8abb-6c9c20149e9f.png)

From there you should see two different property selectors. On your use case, you should set the _Property Name_ to name id while you set _Property Display Name_ to name. This will keep your selector name base, with the full name inside a parameter $neodash_name_nameid_display while having the nameId value on $neodash_name_nameid.

You should be able to play around from there.

I hope I did explain myself :D

**Comment by gcthys** [comment]
Hi @BennuFire ,
Indeed you did and it works fine. Thanks for your patience!

**Comment by gcthys** [comment]
I have a follow-up question concerning Report Actions.
Before, I would set the parameter $neodash_name_name to a value on the same row.
Now (since I followed your advice above), I theoretically have 2 options, see the attachments: I can set
- the Property Name ($neodash_name_nameid) to the content of the ID-cell (P374) , but then the queries in none of the reports return any data
- the Property Display Name ($neodash_name_name) to the content of the `Preferred Name`-cell (HIERONYMUS VERDUSSEN), but then the cell becomes unclickable.

What am I missing? Thanks again.

Screenprints:
![Schermafdruk 2023-02-08 13 14 29](https://user-images.githubusercontent.com/68611501/217527603-ea4c1fc3-a8d7-48d7-8660-3cb5cc62c131.png)
![Schermafdruk 2023-02-08 13 15 18](https://user-images.githubusercontent.com/68611501/217527619-e317826c-c8a3-400d-a550-1e32c9d5c616.png)
![Schermafdruk 2023-02-08 13 15 03](https://user-images.githubusercontent.com/68611501/217527621-35ad8527-908d-4655-b28c-13d47da6fad4.png)
![Schermafdruk 2023-02-08 13 14 49](https://user-images.githubusercontent.com/68611501/217527623-b871dcdb-67bf-4cb3-8c8c-78886cece5be.png)

**Comment by BennuFire** [comment]
Hi @gcthys,

With this implementation you are not really using the display/value option on the parameter select but it's still a valid solution. Here you are talking about an issue with actions. When you set the variable $neodash_name_name with both cell and To as preferred name, is there anything funky on the console? It's clickable because I see the blue button. What's the value on the parameter after you click (when you check the neodash json)?

**Comment by gcthys** [comment]
Hi @BennuFire ,
I hope this is the information you asked for.
This is the code for the two reports: 
"SEARCH NAME" is the main Parameter Select report where I am using the Parameter Display Name. "Producer Found" is the one where the Report Action is set.
`
         "title": "SEARCH NAME",
          "query": "MATCH (n:`Name`) \nWHERE toLower(toString(n.`Name`)) CONTAINS toLower($input) \nRETURN DISTINCT n.`NameID` as value,  n.`Name` as display ORDER BY size(toString(value)) ASC LIMIT 200",
          "width": 3,
          "height": 1,
          "x": 0,
          "y": 0,
          "type": "select",
          "selection": {},
          "settings": {
            "type": "Node Property",
            "entityType": "Name",
            "propertyType": "NameID",
            "backgroundColor": "#FFFFCC",
            "suggestionLimit": 200,
            "overridePropertyDisplayName": true,
            "parameterName": "neodash_name_nameid",
            "propertyTypeDisplay": "Name"
          }
        },
          }
        },
        {
          "title": "Producer Found",
          "query": "MATCH (p:Producer)-[HAS_NAME_VARIANT {qualification: 'preferred'}]->(n:Name)\nWHERE p.ProducerID = $neodash_producer_producerid\nRETURN p.ProducerID AS ID, n.Name AS `Preferred Name`\n",
          "width": 3,
          "height": 1,
          "x": 7,
          "y": 4,
          "type": "table",
          "selection": {},
          "settings": {
            "textAlign": "center",
            "verticalAlign": "middle",
            "fontSize": 32,
            "backgroundColor": "#86b0f2",
            "actionsRules": [
              {
                "condition": "Click",
                "field": "Preferred Name",
                "value": "Preferred Name",
                "customization": "set variable",
                "customizationValue": "name_name"
              }
            ],`
Before I click on the blue button, the relevant parameters are:
`      "neodash_names_name": null,
      "neodash_names_naam": null,
      "neodash_name_name": "Hendrik Alssens",
      "neodash_name_text_2": "Joannes Moretus",
      "neodash_name_nameid": "N75",
      "neodash_name_nameid_display": "Abraham van Dormale",`
After clicking, they are:
`	  "neodash_names_name": null,
      "neodash_names_naam": null,
      "neodash_name_name": "Claes de Grave",
      "neodash_name_text_2": "Joannes Moretus",
      "neodash_name_nameid": "N75",
      "neodash_name_nameid_display": "Abraham van Dormale",`
 where "Abraham van Dormale" and "P75" are values set by "SEARCH NAME" before clicking the blue button. "Claes de Grave" is the name in the blue button in "Producer Found"
